test/utils.test.sh

```diff
@@ -390,6 +390,60 @@ assert "contains" "${log}" "pushing test to us-east-1"
 assert "contains" "${log}" "found existing image for test in us-west-2, skipping push"
 assert "equal" "${FAILURE}" "" "should not have any failures"
 
+# bucket_push() no ImageBucketPrefix
+tag_test "bucket_push"
+export ImageBucketPrefix=""
+
+log=$(bucket_push)
+expected="nothing to do"
+assert "equal" "${log}" "${expected}"
+
+# bucket_push()
+export ImageBucketPrefix="something"
+export bucket_regions=(region1)
+export tmpdir="temp"
+export repo="repo"
+export after="1"
+
+function aws(){
```

Comment:
is this what mocking looks like in bash? 😀

Reply:
indeed it is :)